Docs for cluster service flows

alexeykazakov opened the pull request.

nurali-techie commented on docs/resources/new_cluster.plantuml:
```diff
@@ -0,0 +1,24 @@
+
+user -> usersettings : Add cluster API URL
+group link OSIO - OSD accounts
+	usersettings -> auth : GET /token/link?for={clusterAPI}
```
It seems that there is no response coming back for this call.

nurali-techie commented on docs/resources/new_cluster.plantuml:
```diff
@@ -0,0 +1,24 @@
+	participant "Cluster Service" as cs
+	participant "Auth Service" as auth
+	participant "UI\nuser settings page" as usersettings
+end box
```
I got little confused to understand when req/res (req from left to right and res from right to left) calls are running in other direction than usual.  I see that if we flip position of `OSD Cluster` and `OSIO` boxes then overall seq_flow will become easy to navigate and understand.

alexeykazakov replied:
Good point. I actually created one diagram with all three flows then split them in three without reordering the boxes which made it look weird.

nurali-techie commented on docs/resources/new_cluster.plantuml:
```diff
@@ -0,0 +1,24 @@
+	auth --> cluster
+	cluster --> user : autenticate
+	cluster --> auth : {user_token}
+	auth -> cs : Create new identity_cluster\nPOST /clusters/identity
```
It will help if we add payload details for this call.  As I got, it would be cluster, user_token but not quite sure.

nurali-techie commented on docs/resources/configure_cluster.plantuml:
```diff
@@ -0,0 +1,25 @@
+operator <-- cluster : SA token
+|||
+operator -> auth : obtain OSIO operator Service Account token
+operator <-- auth : {operator_sa_token}
```
Can you plz add here, what auth_mechanism allow auth_service to return operator_sa_token to osio_operator ??

alexeykazakov replied:
It's https://fabric8-services.github.io/fabric8-auth/reference.html#_service_account_authentication
This flow is a very first draft. We will add more details later.

nurali-techie commented on docs/resources/new_user.plantuml:
```diff
@@ -0,0 +1,14 @@
+
+rhduser -> oso : sign up
+oso -> auth : create user (cluster URL)
+auth -> cs : Create new identity_cluster\nPOST /clusters/identity
```
payload details will help.

nurali-techie commented:
@alexeykazakov overall looks good.  I really like the box (with color coding), it makes it super easy to understand the flow :+1:

alexeykazakov commented:
Thanks for the review @nurali-techie! I've addressed some comments. I created this doc some time ago just as a very highlighted flow of cluster service interaction. So, I'm basically just moving it to GH where we can improve it overtime. Some parts have been already implemented but some parts are still in draft version and will need clarification and detalization.
